discovery/pkg/apigee/provision.go

```diff
@@ -42,8 +42,9 @@ type client interface {
 	GetAppCredential(appName, devID, key string) (*models.DeveloperAppCredentials, error)
 	CreateAppCredential(appName, devID string, products []string, expDays int) (*models.DeveloperApp, error)
 	RemoveAppCredential(appName, devID, key string) error
-	AddProductCredential(appName, devID, key string, cpr apigee.CredentialProvisionRequest) (*models.DeveloperAppCredentials, error)
-	RemoveProductCredential(appName, devID, key, productName string) error
+	AddCredentialProduct(appName, devID, key string, cpr apigee.CredentialProvisionRequest) (*models.DeveloperAppCredentials, error)
+	RemoveCredentialProduct(appName, devID, key, productName string) error
```

Comment:
The method is not being used anymore, so do we still need that defined?
Also, since we are now doing an update on the api product to revoke and approve the credential key, the product will never be deleted even after all the credential, access request have been deprovisioned, is this expected?

Reply:
It is expected. I think the functionality is the same, regardless if its removed. This being a better outcome than a cred that is able to be used on all proxies, when no product is on the cred.

I can remove it, but its possible that we might want to add this back, so was hesitant to remove the code.